### PR TITLE
FBP

### PR DESCRIPTION
**Pull request by gfardell**
Update all FBP GPU based algorithms to use new geometry

Create `FBP_Flexible` DataProcessor
Rename `FDK` to `FDK_Flexible` DataProcessor
Rename `FBP` to `FBP_Simple` DataProcessor

Create `FBP` DataProcessor that selects correct sub-class.

for GPU data will use either FDK_Flexible or FBP_Simple
for cpu will use FBP_Simple 2D versions

ToDo: Multi-channel version

### CODE REVIEW

**Comment by gfardell** [comment]
Need to add unittests for FBP

**Comment by gfardell** [comment]
GitHub has got confused.
FBP has been renamed to FBP_simple (not implemented by me, currently only uses 2D parallel cpu version from here)
FBP is a new super processor to do all the delegation to the right FBP/FDK processor.

**Comment by gfardell** [comment]
Need to make source distance >> pixelsize in FBP hack

**Comment by gfardell** [comment]
I believe this is ready to merge pending jenkins

File: Wrappers/Python/ccpi/astra/processors/FBP_Flexible.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        else:
+            detector_direction_col = sino_geom_cone.config.system.detector.direction_col
+            tmp = AcquisitionGeometry.create_Cone3D(cone_source, detector_position, detector_direction_row, detector_direction_col)
+
+        sino_geom_cone.config.system = tmp.config.system.copy()
+
+        vol_geom_astra, proj_geom_astra = convert_geometry_to_astra_vec(volume_geometry, sino_geom_cone)
+ 
+
+        super(FDK_Flexible, self).__init__( volume_geometry = volume_geometry,
```
Comment (paskino) [inline Wrappers/Python/ccpi/astra/processors/FBP_Flexible.py:48]: I suggest to initialise the parent class and then modify the members that should be available.

**Review by paskino — CHANGES_REQUESTED** [review]

**Comment by gfardell** [comment]
I believe this is ready to merge pending jenkins

**Pull request merged**